src/esp/metadata/managers/DatasetAttributesManager.cpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "DatasetAttributesManager.h"
+
+#include "esp/io/io.h"
+#include "esp/io/json.h"
+
+namespace esp {
+namespace metadata {
+
+using attributes::DatasetAttributes;
+namespace managers {
+DatasetAttributes::ptr DatasetAttributesManager::createObject(
+    const std::string& datasetHandle,
+    bool registerTemplate) {
+  std::string msg;
+  DatasetAttributes::ptr attrs = this->createFromJsonOrDefaultInternal(
+      datasetHandle, msg, registerTemplate);
+
+  if (nullptr != attrs) {
+    LOG(INFO) << msg << " dataset attributes created"
+              << (registerTemplate ? " and registered." : ".");
+  }
+  return attrs;
+}  // DatasetAttributesManager::createObject
+
+DatasetAttributes::ptr DatasetAttributesManager::initNewObjectInternal(
+    const std::string& datasetFilename) {
+  DatasetAttributes::ptr newAttributes = this->constructFromDefault();
+  if (nullptr == newAttributes) {
+    newAttributes =
+        DatasetAttributes::create(datasetFilename, physicsAttributesManager_);
+  }
+  // attempt to set source directory if exists
+  this->setFileDirectoryFromHandle(newAttributes);
+
+  // set the handle of the physics manager that is used for this newly-made
+  // dataset
+  newAttributes->setPhysicsManagerHandle(physicsManagerAttributesHandle_);
+  // any internal default configuration here
+  return newAttributes;
+}  // DatasetAttributesManager::initNewObjectInternal
+
+void DatasetAttributesManager::setValsFromJSONDoc(
+    attributes::DatasetAttributes::ptr dsAttribs,
+    const io::JsonGenericValue& jsonConfig) {
+  // dataset root directory to build paths from
+  std::string dsDir = dsAttribs->getFileDirectory();
+  // process stages
+  readDatasetJSONCell(dsDir, "stages", jsonConfig,
+                      dsAttribs->getStageAttributesManager());
+
+  // process objects
+  readDatasetJSONCell(dsDir, "objects", jsonConfig,
+                      dsAttribs->getObjectAttributesManager());
+
+  // process light setups - implement handling light setups TODO
+  // readDatasetJSONCell(dsDir,"light setups", jsonConfig,
+  //                     dsAttribs->getLightsAttributesManager());
+
+  // process scene instances - implement handling scene instances TODO
+  // readDatasetJSONCell(dsDir,"scene instances", jsonConfig,
+  //                     dsAttribs->getSceneInstanceManager());
```

Comment:
These will be made available in the next PR where lights and scene instances will be supported.